test/val/val_decoration_test.cpp

```diff
@@ -20,16 +20,34 @@
 #include "gmock/gmock.h"
 #include "source/val/decoration.h"
 #include "test/unit_spirv.h"
+#include "test/val/val_code_generator.h"
 #include "test/val/val_fixtures.h"
 
 namespace spvtools {
 namespace val {
 namespace {
 
+using ::testing::Combine;
 using ::testing::Eq;
 using ::testing::HasSubstr;
+using ::testing::HasSubstr;
+using ::testing::Values;
+
+struct TestResult {
+  TestResult(spv_result_t in_validation_result = SPV_SUCCESS,
+             const char* in_pre_error_str = nullptr,
+             const char* in_post_error_str = nullptr)
+      : validation_result(in_validation_result),
+        pre_error_str(in_pre_error_str),
+        post_error_str(in_post_error_str) {}
+  spv_result_t validation_result;
+  const char* pre_error_str;
```

Comment:
Is there a good reason to distinguish these strings? No test uses both. They also result in the same check getting performed.

I'd also prefer using std::string over char*, but its just a preference.